binary_classification/src/lib.rs

```diff
@@ -1,90 +1,156 @@
+            "Classify each element in a tensor depending on whether they are above or below a certain threshold.",
+        );
+        metadata.set_repository(env!("CARGO_PKG_REPOSITORY"));
+        metadata.set_homepage(env!("CARGO_PKG_HOMEPAGE"));
+        metadata.add_tag("classify");
+
+        let threshold = ArgumentMetadata::new("threshold");
+        threshold.set_default_value("0.5");
+        threshold.set_description("The classification threshold.");
+        let hint = runtime_v1::supported_argument_type(ArgumentType::Float);
+        threshold.add_hint(&hint);
+        metadata.add_argument(&threshold);
+
+        let input = TensorMetadata::new("input");
+        input.set_description("The numbers to classify");
+        let hint =
+            supported_shapes(&[ElementType::F32], DimensionsParam::Fixed(&[0]));
+        input.add_hint(&hint);
+        metadata.add_input(&input);
+
+        let output = TensorMetadata::new("classified");
+        output.set_description("A tensor of `1`'s and `0`'s, where `1` indicates an element was above the `threshold` and `0` means it was below.");
+        let hint =
+            supported_shapes(&[ElementType::U32], DimensionsParam::Fixed(&[0]));
+        output.add_hint(&hint);
+        metadata.add_output(&output);
+
+        register_node(&metadata);
+    }
 
-impl BinaryClassification {
-    pub const fn new() -> Self { BinaryClassification { threshold: 0.5 } }
-}
+    fn graph(node_id: String) -> Result<(), GraphError> {
+        let ctx = GraphContext::for_node(&node_id)
+            .ok_or(GraphError::MissingContext)?;
+
+        ctx.add_input_tensor(
+            "input",
+            ElementType::U32,
+            DimensionsParam::Fixed(&[0]),
+        );
+        ctx.add_output_tensor(
+            "classified",
+            ElementType::U32,
+            DimensionsParam::Fixed(&[0]),
+        );
+
+        Ok(())
+    }
 
-impl Default for BinaryClassification {
-    fn default() -> Self { BinaryClassification::new() }
+    fn kernel(node_id: String) -> Result<(), KernelError> {
+        let ctx = KernelContext::for_node(&node_id)
+            .ok_or(KernelError::MissingContext)?;
+
+        let threshold = get_threshold(|n| ctx.get_argument(n))
+            .map_err(KernelError::InvalidArgument)?;
+
+        let TensorResult {
+            element_type,
+            dimensions,
+            buffer,
+        } = ctx.get_input_tensor("input").ok_or_else(|| {
+            KernelError::InvalidInput(InvalidInput {
+                name: "input".to_string(),
+                reason: BadInputReason::NotFound,
+            })
+        })?;
+
+        let output = match element_type {
+            ElementType::F32 =>{
+                buffer.view::<f32>(&dimensions)
+                .map_err(|e| KernelError::InvalidInput(InvalidInput{ name: "bounding_boxes".to_string(), reason: BadInputReason::InvalidValue(e.to_string()) }))?;
+                transform(buffer.elements(), threshold)
```

Comment:
Did you mean to assign the result of `buffer.view()` to a variable so it can be passed to `transform()`?